core/src/main/java/com/linecorp/armeria/common/DefaultRequestId.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+import javax.annotation.Nullable;
+
+/**
+ * The default {@link RequestId} implementation.
+ */
+final class DefaultRequestId implements RequestId {
+
+    private static final byte[] HEXDIGITS = { '0', '1', '2', '3', '4', '5', '6', '7',
+                                              '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
+
+    private final long value;
+
+    @Nullable
+    private String longText;
+    @Nullable
+    private String shortText;
+
+    DefaultRequestId() {
+        this(ThreadLocalRandom.current().nextLong());
+    }
+
+    DefaultRequestId(long value) {
+        this.value = value;
+    }
+
+    @Override
+    public String text() {
+        if (longText != null) {
+            return longText;
+        }
+
+        return longText = newLongText();
+    }
+
+    private String newLongText() {
+        final String newLongText;
+        if (Integer.numberOfLeadingZeros((int) (value >>> 32)) < 4) {
```

Comment:
Think it's worth adding a small comment on why we need the slow branch or what would go wrong if we didn't.

Reply:
Added comments and optimized it a little bit more.